File: CMOWebApi.Models/ComparetiveModel/ComparativeTargetEntryModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CMOWebApi.Models.ComparetiveModel
{
	public class ComparativeTargetEntryModel
	{

		public int Id { get; set; }
		public Nullable<int> Code { get; set; }
		public Nullable<int> DepartmentCode { get; set; }
		public Nullable<int> YearCode { get; set; }
		public Nullable<int> KPICategoryCode { get; set; }
		public bool IsActive { get; set; }
		public bool IsDeleted { get; set; }
		public Nullable<System.DateTime> CreatedDate { get; set; }
		public Nullable<int> CreatedBy { get; set; }
		public Nullable<System.DateTime> ModifiedDate { get; set; }
		public Nullable<int> ModifiedBy { get; set; }
		public List<ComparativeParameterTargetParameterMappingModel> ComparativeParameterTargetParameterMappingModel { get; set; }
	}

	public class ComparativeParameterTargetParameterMappingModel
	{
		public long Id { get; set; }
		public Nullable<int> ComparativeParameterTargetId { get; set; }
		public Nullable<long> ParameterCode { get; set; }
        public string PhysicalParameter { get; set; }
        public string FinancialParameter { get; set; }
        public string physicalUnitName { get; set; }
        public string FinancialUnitName { get; set; }
        public string PhysicalValue { get; set; }
		public string FinancialValue { get; set; }
        public Nullable<bool> IsFinancial { get; set; }
        public Nullable<bool> IsPhysical { get; set; }
		public string KPICategoryName { get; set; }
		public Nullable<int> KPICategoryCode { get; set; }
	}

	public class ComparativeParameterTargetViewModel
	{
		public int Id { get; set; }
		public Nullable<int> Code { get; set; }
		public Nullable<int> DepartmentCode { get; set; }
		public Nullable<int> KPICategoryCode { get; set; }
		public bool IsActive { get; set; }
		public bool IsDeleted { get; set; }
		public string DepartmentTitle { get; set; }
		public string KPICategoryName { get; set; }
		public string YearName { get; set; }
		public Nullable<System.DateTime> CreatedDate { get; set; }
		public Nullable<System.DateTime> ModifiedDate { get; set; }
		public string modifiedbyName { get; set; }
	}

	public class ComparativeTargetParmeterListModel
	{
        public long ParameterMappingId { get; set; }
        public string PhysicalValue { get; set; }
        public string FinancialValue { get; set; }
        public Nullable<long> ParameterCode { get; set; }
		public string PhysicalParameter { get; set; }
		public string FinancialParameter { get; set; }
		public string DepartmentTitle { get; set; }
		public string KPICategoryName { get; set; }
		public string physicalUnitName { get; set; }
		public string FinancialUnitName { get; set; }
        public Nullable<bool> IsFinancial { get; set; }
        public Nullable<bool> IsPhysical { get; set; }
    }
}

File: ANGULAR/ANGULAR/src/app/Shared/Model/Camparetive/comparative-target-entry-model.ts
export class ComparativeTargetEntryModel {

		Id: number;
		Code: number | null;
		DepartmentCode: number | string | null;
		YearCode: number | string | null;
		KPICategoryCode: number | string | null;
		IsActive: boolean;
		IsDeleted: boolean;
		CreatedDate: Date | string | null;
		CreatedBy: number | null;
		ModifiedDate: Date | string | null;
		ModifiedBy: number | null;
		ComparativeParameterTargetParameterMappingModel: ComparativeParameterTargetParameterMappingModel[] = [];
	}

	export class ComparativeParameterTargetParameterMappingModel {
		Id: number;
		ComparativeParameterTargetId: number | null;
		ParameterCode: number | string | null;
		PhysicalValue: string;
    FinancialValue: string;
    IsFinancial: boolean;
    IsPhysical: boolean;
    KPICategoryName: string;
    constructor(){
      this.ComparativeParameterTargetId=0;
      this.Id=0;
    }
  }


  export class ComparativeParameterTargetViewModel {
		Id: number;
		IsActive: boolean;
		DepartmentTitle: string;
		KPICategoryName: string;
		YearName: string;
		ModifiedDate: Date | string | null;
    modifiedbyName: string;
  }

  export class ComparativeTargetParmeterListModel {
		Id: number;
    ParameterMappingId: number;
    PhysicalValue: string;
    FinancialValue: string;
    ParameterCode: number | null;
		PhysicalParameter: string;
		FinancialParameter: string;
		DepartmentTitle: string;
		KPICategoryName: string;
		physicalUnitName: string;
    FinancialUnitName: string;
    ComparativeParameterTargetId: number | null;
    IsFinancial: boolean;
    IsPhysical: boolean;
	}
